CMakeLists.txt

```diff
@@ -25,13 +25,19 @@ endif()
 add_library(tiny-process-library ${process_source_files})
 
 target_link_libraries(tiny-process-library ${CMAKE_THREAD_LIBS_INIT})
+target_include_directories(tiny-process-library PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
I notice that this line is not necessary even though tiny-process-library is a sub-project. Although one would have to add the tiny-process-library header path through `include_directories` in the main project. I guess this is common practice though. Is it ok that we remove this line?

Reply:
No, it isn't. You should just be able to write `target_link_libraries` and everything works.

Reply:
Yes it is. Unless you have target_include_directories(... PUBLIC) the include path will not be propagated to the dependent target via target_link_libraries.

Reply:
include_directories is the OLD way to propagate directories and is global for ALL targets.

Reply:
target_include_directories have been in CMake since version 2.8.12.

Reply:
Thank you guys, keeping it then:)